core/src/main/java/org/apache/metamodel/query/FromItem.java

```diff
@@ -267,6 +267,11 @@ private void appendJoinOnItem(StringBuilder sb, String sideAlias, SelectItem onI
             sb.append(superQueryAlias);
             return;
         }
+        
+        if(_join != null && _leftSide.getJoin() != null) {
```

Comment:
And actually this trick avoids/skips the sideAlias thing ... might be a bit too destructive. Do we _need_ to avoid the parenthesis or is it just cosmetic?